app.rb

```diff
@@ -20,22 +26,43 @@
 
 CACHE_EXPIRY_TIMEOUT = 60 # How long a room events API hit should cache, in seconds
 
-ROOMS = JSON.parse(File.read("rooms.json")).map { |e|
-  [e.fetch("slug").to_sym, Room.new(
-    name: e.fetch("name"),
-    css_class: e.fetch("css_class"),
-    gcal_identifier: e.fetch("gcal_identifier"),
-    presence_colour_rgb: e.fetch("presence_colour_rgb", [255, 255, 255])
-  )]
-}.to_h.freeze
-
-BOARDS = JSON.parse(File.read("boards.json")).map { |e|
-  [e.fetch("slug").to_sym, {
-    name: e.fetch("name"),
-    rooms: e.fetch("rooms").map { |e| ROOMS[e.to_sym] },
-    show_clock: e.fetch("show_clock", false)
-  }]
-}.to_h.freeze
+ROOMS = if ENV["RACK_ENV"] == "test"
+  JSON.parse(File.read("./spec/test_rooms.json")).map { |e|
```

Comment:
These feel like quite a chunk of duplication - it might be cleaner to do something like

```
if ENV["RACK_ENV"] == "test"
    rooms_json_path = "./spec/test_rooms.json"
    boards_json_path = "./spec/test_boards.json"
else
    rooms_json_path = "rooms.json"
    boards_json_path = "boards.json"
end
```

and then using the `rooms_json_path` and `boards_json_path` variables in the `File.read` operations. That way you're reducing the duplication, and moving the question of "where does this information come from?" outside of the bit of code which does something with it.

Reply:
Your way definitely looks cleaner and easier to read. I'll make those changes before merging. Thank you Nick!